### PR TITLE
Ywh114/refactor

### PR DESCRIPTION
**Pull request by ywh114**
- Cleaned up extensively
- Moved most functionality into modules
- Created distribution system via executable zips

### CODE REVIEW

**Comment by ywh114** [comment]
Readme needs to be changed again :/

**Review by ziyiwang54 — APPROVED** [review]
Looks good! Ready to merge.

File: scripts/python/multitest.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            if keyboard.is_pressed('q'):
+                motor2.set_state_send(W=MOTOR2_SPEED)
+            # small arm motor backward
+            elif keyboard.is_pressed('a'):
+                motor2.set_state_send(W=-MOTOR2_SPEED)
+            else:
+                motor2.set_state_send(W=0, K_W=20)
+
+            # No output if all 0s
+            if any(motor.get_state_send['W'] for motor in motors):
```
Comment (ywh114) [inline scripts/python/multitest.py:85]: Invalid. -> .get_state_send()['W']

File: scripts/python/util/libman.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        tup = ('Win', '64')
+    elif system == 'Linux':
+        # x86_64
+        if machine == 'x86_64':
+            tup = ('Linux', '64')
+        # i?86
+        elif machine.endswith('86'):
+            tup = ('Linux', '32')
+        # aarch32/64
+        elif uname.machine.startswith('aarch'):
```
Comment (ywh114) [inline scripts/python/util/libman.py:56]: Unnecessary uname

File: scripts/python/util/libman.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        with ZipContextLibHandler(DISTINFO.ROOT_DIR, so_path):
+            cdll = T.cdll.LoadLibrary(so_path)
+    else:
+        cdll = T.cdll.LoadLibrary(so_path)
+
+    return cdll
+
+def cdll_init(so_path):
+    """Returns a ctypes.cdll from a shared object path
+    Initializes it with presets"""
```
Comment (ywh114) [inline scripts/python/util/libman.py:79]: Initialized

File: scripts/python/util/motor.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        state['mode'] = motor.mode
+        state['T'] = motor.T
+        state['W'] = motor.W
+        state['Pos'] = motor.Pos
+        state['K_P'] = motor.K_P
+        state['K_W'] = motor.K_W
+
+    def set_state_send(self, **new_state):
+        """Set the state_send dictionary and update MOTOR_send"""
+        # [s]elf [k]eys, [n]ew $_, [e]xisting $_
```
Comment (ywh114) [inline scripts/python/util/motor.py:50]: Useless comment [e]xist...

File: scripts/python/util/motor.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        self.state |= {k: v
+                       for k, v in new_state.items()
+                       if k in existing_keys}
+        self.update_motor_from_state(self.motor_send, self.state_send)
+
+    def send(self):
+        """Send the current MOTOR_send to fd"""
+        self.cdll.modify_data(Types.byref(self.motor_send))
+        self.cdll.send_recv(self.fd, Types.byref(self.motor_send),
+                                     Types.byref(self.motor_recv))
```
Comment (ywh114) [inline scripts/python/util/motor.py:64]: nit: use better alignment

File: scripts/python/util/ports.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Illinois RoboMaster 2024
+import os
+import platform
+
+def get_port(bytes_=False):
+    """Searches for possible ports and returns the first one found"""
+    not_found = 'Cannot open serial: device not attatched.'
```
Comment (ywh114) [inline scripts/python/util/ports.py:7]: nit: variable name should reflect that it is a message

**Review by ywh114 — COMMENTED** [review]
Make small fixes

**Pull request merged**